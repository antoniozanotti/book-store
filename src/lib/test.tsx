import React, { ReactElement, ReactNode } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return <div data-testid="container">{children}</div>;
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";
import { usePathname } from "next/navigation";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
    notFound: () => null,
    usePathname: () => null,
    useSearchParams: () => null,
  };
});
