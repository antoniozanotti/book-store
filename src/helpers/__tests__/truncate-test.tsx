import { render, screen } from "@/lib/test";
import { truncate } from "../truncate";

describe("truncate", () => {
  it("should truncate text", async () => {
    const value = truncate("Antonio Zanotti", 5);
    expect(value).toMatchSnapshot();
  });

  it("should not truncate text", async () => {
    const value = truncate("Antonio Zanotti", 50);
    expect(value).toMatchSnapshot();
  });
});