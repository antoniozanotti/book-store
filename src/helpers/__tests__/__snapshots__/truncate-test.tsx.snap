// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`truncate should not truncate text 1`] = `"Antonio Zanotti"`;

exports[`truncate should truncate text 1`] = `"An..."`;
